mkinitcpio

```diff
@@ -251,6 +262,92 @@ build_image() {
     fi
 }
 
+build_uefi(){
+    local out=$1 initramfs=$2 cmdline=$3 osrelease=$4 splash=$5 kernelimg=$6 uefistub=$7 microcode=(${@:7}) errmsg=
+    OBJCOPYARGS=()
+
+    msg "Creating UEFI executable: %s" "$out"
+
+    if [[ -z "$uefistub" ]]; then
+        for stub in {/usr,}/lib/{systemd/boot/efi,gummiboot}/linux{x64,ia32}.efi.stub; do
+            if [[ -f "$stub" ]]; then
+                uefistub="$stub"
+                msg2 "Using UEFI stub: %s" "$uefistub"
+                break
+            fi
+        done
+    elif [[ ! -f "$uefisub" ]]; then
+        error "UEFI stub '%s' not found" "$uefistub"
+        return 1
+    fi
+
+    if [[ -z "$kernelimg" ]]; then
+        for img in "/lib/modules/$KERNELVERSION/vmlinuz" "/boot/vmlinuz-$KERNELVERSION" "/boot/vmlinuz-linux"; do
+            if [[ -f "$img" ]]; then
+                kernelimg="$img"
+                msg2 "Using kernel image: %s" "$kernelimg"
+                break
+            fi
+        done
+    fi
+    if [[ ! -f "$kernelimg" ]]; then
+        error "Kernel image '%s' not found" "$kernelimage"
+        return 1
+    fi
+
+    if [[ -z "$cmdline" ]]; then
+        if [[ -f "/etc/kernel/cmdline" ]]; then
+            cmdline="/etc/kernel/cmdline"
+        elif [[ -f "/usr/lib/kernel/cmdline" ]]; then
+            cmdline="/usr/lib/kernel/cmdline"
+        else
+            warning "Note: /etc/kernel/cmdline does not exist and --cmdline is unset!"
+            cmdline="/proc/cmdline"
+            warning "Reusing current kernel cmdline from $cmdline"
+        fi
+        msg2 "Using cmdline file %s" "$cmdline"
+    fi
+    if [[ ! -f "$cmdline" ]]; then
```

Comment:
Either this condition is wrong, or the description of the `--cmdline` option is misleading.

> Set kernel line (default content: /etc/kernel/cmdline, /proc/cmdline)

I thought that I can just directly pass the kernel line, but it actually expects a file.

Reply:
It's expecting a file so it should *probably* mention it, but you could just pass `--cmdline < <(echo "rw root=/dev/mapper/root")` or something similar i reckon.

Reply:
Not probably, but definitely; it’s very unclear now. ;)

Process substitution (`<( cmd )`) is not defined by POSIX command language, nor supported by majority of the shells, just by Bash and ZSH, so it’s not portable. Sadly, mkinitcpio contains a lot of bashisms, but one must have Bash installed to use it, but that doesn’t mean it will be always called from Bash.

Now I have to do the following:

```sh
cmdline_file=$(mktemp)
printf %s "$cmdline" > "$cmdline_file"

mkinitcpio --cmdline "$cmdline_file" ...

rm -f "$cmdline_file"
```

…which is suboptimal.

Allowing to just pass cmdline as a value would be IMHO more convenient in general.

Reply:
> Allowing to just pass cmdline as a value would be IMHO more convenient in general. Just a suggestion.

I guess that depends what you do with mkinitcpio.

I've almost never wanted to specify it as an argument, as opposed to having it in /etc/kernel/cmdline

Reply:
Supporting this as a string when dealing with the preset files and stuff would just be tedious and a lot harder to implement correctly. I'm happy with this being a file and this is what other initramfs generators following `kernel-install` and `dracut` does as well.